reporters/cloud-run-scripts/CloudResponse.js

```diff
@@ -1,54 +1,68 @@
-  // to be passed in the req body and then set as environment variables in the child_process as a string
-  let processedEnvironmentVariables = {};
-
-  if (req.body.environmentVariables) {
-    Object.keys(req.body.environmentVariables).forEach(key => {
-      // All env variables must be a string. If they are not a string (int, object ect) convert them to a string.
-      processedEnvironmentVariables[key] =
-        typeof req.body.environmentVariables[key] == "string"
-          ? req.body.environmentVariables[key]
-          : JSON.stringify(req.body.environmentVariables[key]);
-    });
-  }
+    // Iterate over the provided environment variables and ensure that they are all strings. This enables json configs
+    // to be passed in the req body and then set as environment variables in the child_process as a string
+    let processedEnvironmentVariables = {};
 
-  // Run the command from the request body. Note this assumes that the process is running from the /core directory.
-  // Include the environment variables. Having both ...process.env and ...processedEnvironmentVariables acts to combined
-  // the existing env variables with those passed in from the req.
-  exec(
-    req.body.cloudRunCommand,
-    { env: { ...process.env, ...processedEnvironmentVariables } },
-    (error, stdout, stderr) => {
-      if (error !== null) {
-        console.error(`exec error: ${error}stderr: ${stderr}`);
-        res.status(400).send({
-          message: "ERROR executing process!",
-          stdout,
-          stderr,
-          error
-        });
-      } else {
-        console.error(`stdout: ${stdout}`);
-        res.status(200).send({ message: "Process executed correctly!", stdout, stderr, error: null });
-      }
+    if (req.body.environmentVariables) {
+      Object.keys(req.body.environmentVariables).forEach(key => {
+        // All env variables must be a string. If they are not a string (int, object ect) convert them to a string.
+        processedEnvironmentVariables[key] =
+          typeof req.body.environmentVariables[key] == "string"
+            ? req.body.environmentVariables[key]
+            : JSON.stringify(req.body.environmentVariables[key]);
+      });
     }
-  );
+
+    const execResponse = await execShellCommand(req.body.cloudRunCommand, processedEnvironmentVariables);
+    // TODO: refactor this to send a more graceful winston log.
+    console.log("error:", execResponse.error);
+    console.log("stdout:", execResponse.stdout);
+    console.log("stderr:", execResponse.stderr);
+
+    if (execResponse.error) {
+      console.log("Process exited with error");
+      throw execResponse.error;
+    }
+    console.log("Process exited with no error");
+
+    res.status(200).send({ message: "Process exited correctly!", error: null });
+  } catch (error) {
+    res.status(400).send({ message: "Process exited with error!", error: error });
+  }
 });
 
+function execShellCommand(cmd, inputEnv) {
```

Comment:
The child process logic was refined to run as a promise. Previously this was not strictly blocking logic and there was inconsistent behavior with executing multiple in parallel. This new implementation returns a promise that will only resolve when the process exits.